
<launch>
  <!--使用USB_CAMERA 带的相机来检测圆环 来恢复机器人的位置-->
  <node pkg="apriltag_detector" type="apriltag_detector" name="apriltag_detector" output="screen" >

    <!--建议使用模式1-->
    <param name="~High_performance_mode" value= "1" />
    <!--处理频率默认10-->
    <param name="~process_hz" value="10.0" />
    <!--最近两个圆环圆心之间的距离 默认8CM-->
    <param name="~default_tag_size" value="0.075" />
    <!--最近聚类聚类 默认100-->
    <param name="~dbscan_distance" value="100" />
    <!--是否显示调试画面-->
    <param name="~viewer" value="true" />
    <!--是否显示调试信息-->
    <param name="~debug" value="false" />

    <!--相机的topic-->
    <param name="camera_topic" value="/usb_cam/image_raw" />
    <!--相机的参数topic-->
    <param name="camera_info" value="/usb_cam/camera_info" />
    <!--是否显示UI-->
    <param name="~display_type" value="CUBE" />
    <param name="~marker_thickness" value="0.02" />
    <param name="~display_marker_axes" value="true" />

    <param name="~display_marker_overlay" value="true" />
    <param name="~display_marker_outline_" value="true" />
    <param name="~display_marker_edges_" value="true" />
    <param name="~display_marker_id_" value="true" />

    <remap from="~image" to="/usb_cam/raw_image"/>
    <remap from="~camera_info" to="/usb_cam/camera_info"/>
    <remap from="~marker_array" to="/apriltags/marker_array"/>
    <remap from="~detections" to="/apriltags/detections"/>
  </node>
</launch>
